components/camel-box/camel-box-api/src/main/java/org/apache/camel/component/box/api/BoxFoldersManager.java

```diff
@@ -189,6 +189,43 @@ public BoxFolder createFolder(String parentFolderId, String folderName) {
     }
 
     /**
+     * Create a folder specified by path from parent folder with given <code>parentFolderId</code>,
+     * creating intermediate directories as required.
+     *
+     * @param parentFolderId
+     *            - the id of parent folder.
+     * @param path
+     *            - Sequence of Box folder names from parent folder to returned
+     *            folder.
+     * @return The last folder in path, no fault will be thrown if it already exists.
+     */
+    public BoxFolder createFolder(String parentFolderId, String... path) {
+        try {
+            LOG.debug("Creating folder with path '" + path + "' in parent_folder(id=" + parentFolderId + ")");
+            if (parentFolderId == null) {
+                throw new IllegalArgumentException("Parameter 'parentFolderId' can not be null");
+            }
+            if (path == null) {
+                throw new IllegalArgumentException("Paramerer 'path' can not be null");
+            }
+            BoxFolder folder = new BoxFolder(boxConnection, parentFolderId);
+            searchPath: for (int folderIndex = 0; folderIndex < path.length; folderIndex++) {
```

Comment:
Can we avoid using a goto via label, as its not as good to maintain - and we don't really use them in the other components.

Reply:
Absolutely, I just handled it the same way as was done previously in the same code base, but I'll revise.

Reply:
Ah okay didnt spot that, but we shouldn't rely on goto's so it would be good to fix anyway.